client/configuration_variable.go

```diff
@@ -120,15 +110,15 @@ func (client *ApiClient) ConfigurationVariablesByScope(scope Scope, scopeId stri
 		return []ConfigurationVariable{}, err
 	}
 
-	// The API returns global and template scopes for environment (and other) scopes. Filter them out.
-	if scope != ScopeGlobal {
-		result = filterOutConfigurationVariables(result, ScopeGlobal)
-		if scope != ScopeTemplate {
-			result = filterOutConfigurationVariables(result, ScopeTemplate)
+	// The API returns variables of upper scopes. Filter them out.
+	var filteredVariables []ConfigurationVariable
+	for _, variable := range result {
+		if scopeId == variable.ScopeId {
```

Comment:
I would also add `scope == variable.Scope` here for safety

Reply:
@Wassap124 - do you concur?